enhancements/cloud-integration/tokenized_auth_enablement_operators_on_cloud.md

```diff
@@ -0,0 +1,527 @@
+
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Guide and assist Day-2 Operator admins in providing the required cloud provider credentials matched to their permission
+needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+[NB: these are all short now, just adding to them as we outline what each involved component will do in the solution]
+
+Cloud Credential Operator (CCO) changes: Add an STS mode, distinct from disabled/manual mode that will look for and
+process CredentialRequests referenced in Operator CRs. This will operate in one of two ways depending on which other parts 
+of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with STS mode or continuing with the
+current support for the pod identity webhook).
+- CCO STS mode flow will work by adding a "role-arn"-type field on operator added CredentialsRequest objects, this will 
+trigger CCO to annotate the ServiceAccount on behalf of the operator.
+- Webhook mode will work by annotating the ServiceAccount triggering the projection of the service account tokens into pods
+ created for the operator (same method and resultant actions as: https://github.com/openshift/hypershift/pull/1351)
+
+HyperShift changes: Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+OperatorHub changes: Allow for import of additional, well-known, ENV variables on the Subscription object. These fields
+will allows UX for the information needed by CCO or webhook, for input of the cloud credentials. 
+
+OLM changes: Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by 
+parsing the Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended
+before an operator upgrade.
+
+Operator team/ Operator SDK changes: Follow new guidelines for allowing for the operator to work on an STS enabled. 
+New guidelines would include changing CRD to add CredentialRequest references and putting those referenced
+[@jharrington22 need help turning these bullet items into concrete steps]
+- CredentialRequests into a defined directory in the bundle.
```

Comment:
could it be embedded in a CRD owned by the operator?

Reply:
I think an embedded reference to them, but the CredentialsRequest should be present so that CCO can list and get them easily.